### PR TITLE
junit test

### PR DESCRIPTION
**Pull request by wang1351**
test junit

### CODE REVIEW

File: src/main/java/seedu/duke/entities/Order.java [hunk 1]
```diff
@@ -1,12 +1,17 @@
...
 
 public class Order {
-    private final ArrayList<Dish> dishes;
+    private ArrayList<Dish> dishes = null;
 
-    Order() {
-        this.dishes = new ArrayList<Dish>();
+    public <T> Order(List<T> asList) {
+        
+    }
```
Comment (darylhjd) [inline src/main/java/seedu/duke/entities/Order.java:11]: why suddenly got generic constructor lol

**Review by darylhjd — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/duke/controllers/OrderController.java [hunk 2]
```diff
@@ -57,7 +57,7 @@ private void addOrder() throws OperationTerminationException {
         while (notQuit) {
             Object inputObj = userInput;
             int size = orderManager.addDishToOrder(inputObj);
-            userInput = InputParser.getString("You’ve already added " + size + "dish(es), some more: \n");
+            userInput = InputParser.getString("You’ve already added " + size + " dish(es), some more: \n");
             if (userInput == "Q" || userInput == "q") {
                 notQuit = false;
             }
```
Comment (darylhjd) [inline src/main/java/seedu/duke/controllers/OrderController.java:63]: user can quit anytime by inputting '-' character when using `InputParser` methods. dunnid add your own.

**Review by darylhjd — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/duke/controllers/OrderController.java (hunk 2 above)
Comment (wang1351) [reply src/main/java/seedu/duke/controllers/OrderController.java:63]: noted with thanks

File: src/main/java/seedu/duke/controllers/OrderController.java [hunk 3]
```diff
@@ -50,17 +50,11 @@ protected boolean optionSwitcher(int choice) throws OperationTerminationExceptio
...
+        String userInput = InputParser.getString("Enter dishes you want to order (enter '-' to exit): ");
         while (notQuit) {
             Object inputObj = userInput;
             int size = orderManager.addDishToOrder(inputObj);
-            userInput = InputParser.getString("You’ve already added " + size + "dish(es), some more: \n");
-            if (userInput == "Q" || userInput == "q") {
-                notQuit = false;
-            }
+            userInput = InputParser.getString("You’ve already added " + size + " dish(es), some more: \n");
         }
```
Comment (darylhjd) [inline src/main/java/seedu/duke/controllers/OrderController.java:58]: can simply be `while (true)`

**Review by darylhjd — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/duke/controllers/OrderController.java (hunk 3 above)
Comment (darylhjd) [reply src/main/java/seedu/duke/controllers/OrderController.java:58]: btw, the gradle tests fail now lol

Comment (wang1351) [reply src/main/java/seedu/duke/controllers/OrderController.java:58]: it seems that the constructor is useful

Comment (darylhjd) [reply src/main/java/seedu/duke/controllers/OrderController.java]: doesnt need to be generic tho, can just be `List<Dish> dishList`

File: src/main/java/seedu/duke/entities/Order.java [hunk 4]
```diff
@@ -1,12 +1,17 @@
 package seedu.duke.entities;
 
 import java.util.ArrayList;
+import java.util.List;
 
 public class Order {
-    private final ArrayList<Dish> dishes;
+    private ArrayList<Dish> dishes = null;
```
Comment (BotBw) [inline src/main/java/seedu/duke/entities/Order.java:7]: no need the `= null` part

**Review by BotBw — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/duke/entities/Order.java [hunk 5]
```diff
@@ -1,12 +1,16 @@
...
 import java.util.ArrayList;
+import java.util.List;
 
 public class Order {
-    private final ArrayList<Dish> dishes;
+    private List<Dish> dishes;
 
-    Order() {
-        this.dishes = new ArrayList<Dish>();
+    public <T> Order(List<T> asList) {
```
Comment (darylhjd) [inline src/main/java/seedu/duke/entities/Order.java:9]: no generics needed

**Pull request merged**